### PR TITLE
build(cmd): upgrade clap to 4.x

### PR DESCRIPTION
**Pull request by bigboss2063**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Since the currently used clap version is 3.x, which is already outdated for a long time, it will be updated to the latest 4.4.8.

Because the `multiple` is deprecated,  I use `num_args = 1..` to replace.
## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
[#2742](https://github.com/GreptimeTeam/greptimedb/issues/2742)

### CODE REVIEW

File: src/cmd/Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ async-trait.workspace = true
 auth.workspace = true
 catalog.workspace = true
 chrono.workspace = true
-clap = { version = "3.1", features = ["derive"] }
+clap = { version = "4.4.8", features = ["derive"] }
```
Comment (WenyXu) [inline src/cmd/Cargo.toml:21]:
```suggestion
clap = { version = "4.4", features = ["derive"] }
```

**Review by WenyXu — APPROVED** [review]

File: src/cmd/Cargo.toml (hunk 1 above)
Comment (bigboss2063) [reply src/cmd/Cargo.toml]: changed

**Review by killme2008 — APPROVED** [review]
LGTM. Thank u.

**Comment by bigboss2063** [comment]
Thanks! Hope to be able to do more with greptimedb.

**Pull request merged**